### PR TITLE
[Issue-3801] Webapp - Integrate off-ramp

### PR DESCRIPTION
**Pull request by tunghp2002**

### CODE REVIEW

File: packages/extension-base/src/services/chain-service/index.ts [hunk 1]
```diff
@@ -453,6 +453,10 @@ export class ChainService {
     return filterAssetsByChainAndType(this.getAssetRegistry(), chainSlug, assetTypes);
   }
 
+  public getAssetByChainAndSymbol (chainSlug: string, symbol: string) {
```
Comment (nampc1) [inline packages/extension-base/src/services/chain-service/index.ts:456]: isn't guaranteed to receive a unique asset

**Review by S2kael — CHANGES_REQUESTED** [review]

File: packages/extension-web-ui/src/Popup/Home/Tokens/index.tsx [hunk 2]
```diff
@@ -55,6 +55,7 @@ const Component = (): React.ReactElement => {
   const { accountBalance: { tokenGroupBalanceMap,
     totalBalanceInfo }, tokenGroupStructure: { sortedTokenGroups } } = useContext(HomeContext);
   const currentAccount = useSelector((state: RootState) => state.accountState.currentAccount);
+  const accounts = useSelector((state: RootState) => state.accountState.accounts);
```
Comment (S2kael) [inline packages/extension-web-ui/src/Popup/Home/Tokens/index.tsx:58]: Remove unused code

File: packages/extension-web-ui/src/Popup/BuyTokens.tsx [hunk 3]
```diff
@@ -373,14 +403,42 @@ function Component ({ className, modalContent, slug }: Props) {
...
+            backgroundColor: '#252525',
+            borderRadius: '0.5rem',
+            transition: 'left 0.3s ease-in-out'
+          }}
+          ></div>
+
+          <div
+            className='__service-selector'
+            // eslint-disable-next-line react/jsx-no-bind
+            onClick={handleBuyForm}
```
Comment (S2kael) [inline packages/extension-web-ui/src/Popup/BuyTokens.tsx:426]: Convert a `useCallback` function

File: packages/extension-web-ui/src/Popup/BuyTokens.tsx [hunk 4]
```diff
@@ -373,14 +403,42 @@ function Component ({ className, modalContent, slug }: Props) {
...
+          >
+              Buy
+          </div>
+          <div
+            className='__service-selector'
+            // eslint-disable-next-line react/jsx-no-bind
+            onClick={handleSellForm}
```
Comment (S2kael) [inline packages/extension-web-ui/src/Popup/BuyTokens.tsx:433]: Convert a `useCallback` function

File: packages/extension-web-ui/src/Popup/BuyTokens.tsx [hunk 5]
```diff
@@ -573,6 +632,26 @@ const BuyTokens = styled(Wrapper)<Props>(({ theme: { token } }: Props) => {
       paddingRight: token.padding
     },
 
+    '.__service-container': {
+      backgroundColor: '#1A1A1A',
```
Comment (S2kael) [inline packages/extension-web-ui/src/Popup/BuyTokens.tsx:636]: Use `token.colorBgSecondary`

File: packages/extension-web-ui/src/Popup/Root.tsx [hunk 6]
```diff
@@ -128,6 +156,17 @@ function DefaultRoute ({ children }: {children: React.ReactNode}): React.ReactEl
     , [accounts]
   );
 
+  useEffect(() => {
```
Comment (S2kael) [inline packages/extension-web-ui/src/Popup/Root.tsx:159]: Improve performance

File: packages/extension-web-ui/src/types/buy.ts [hunk 7]
```diff
@@ -4,6 +4,7 @@
 export interface BuyService {
```
Comment (S2kael) [inline packages/extension-web-ui/src/types/buy.ts:4]: Remove unnecessary interface (duplicated)

File: packages/extension-web-ui/src/utils/buy/transak.ts [hunk 8]
```diff
@@ -5,17 +5,40 @@ import { TRANSAK_API_KEY, TRANSAK_URL } from '@subwallet/extension-web-ui/consta
...
+
+export const createTransakOrder: CreateBuyOrderFunction = (orderParams) => {
+  const { action = 'BUY', address, network, slug = '', symbol } = orderParams;
+
   return new Promise((resolve) => {
-    const params = {
-      apiKey: TRANSAK_API_KEY,
+    const location = window.location.origin;
+    const params: TransakOrderParams = {
+      apiKey: '307807a5-5fb3-4add-8a6c-fca4972e0470',
```
Comment (S2kael) [inline packages/extension-web-ui/src/utils/buy/transak.ts:26]: Rollback

File: packages/extension-web-ui/src/utils/buy/transak.ts [hunk 9]
```diff
@@ -5,17 +5,40 @@ import { TRANSAK_API_KEY, TRANSAK_URL } from '@subwallet/extension-web-ui/consta
...
+}
+
+export const createTransakOrder: CreateBuyOrderFunction = (orderParams) => {
+  const { action = 'BUY', address, network, slug = '', symbol } = orderParams;
+
   return new Promise((resolve) => {
-    const params = {
-      apiKey: TRANSAK_API_KEY,
+    const location = window.location.origin;
+    const params: TransakOrderParams = {
```
Comment (S2kael) [inline packages/extension-web-ui/src/utils/buy/transak.ts:25]: Can use `Record<string, string | number | boolean | null>` instead

**Review by S2kael — CHANGES_REQUESTED** [review]

**Review by S2kael — APPROVED** [review]

**Pull request merged**